core/rpc-servers/Cargo.toml

```diff
@@ -5,10 +5,12 @@ authors = ["Parity Technologies <admin@parity.io>"]
 edition = "2018"
 
 [dependencies]
-http = { package = "jsonrpc-http-server", version = "12.0.0" }
 pubsub = { package = "jsonrpc-pubsub", version = "12.0.0" }
-ws = { package = "jsonrpc-ws-server", version = "12.0.0" }
 log = "0.4"
 serde = "1.0"
 substrate-rpc = { path = "../rpc" }
 sr-primitives = { path = "../sr-primitives" }
+
+[target.'cfg(not(target_os = "unknown"))'.dependencies]
```

Comment:
Maybe introduce a `std` feature and tag these dependencies as `optional`?

Reply:
That would be the ideal way to do it, if it wasn't for https://github.com/rust-lang/cargo/issues/5730.
In substrate-service, I would have to depend on `std` for non-WASM, and not-std for WASM. But because of that Cargo bug, the `std` would always be compiled.

Reply:
Ahh this issue again :)